src/implementations/vanilla/components/select/select.js

```diff
@@ -424,11 +440,31 @@ export class Select {
     });
     // Select all checkbox follows along.
     const checked = visible.filter((c) => c.querySelector('input').checked);
-    if (visible.length !== checked.length) {
+    if (visible.length === 0 || visible.length !== checked.length) {
       this.selectAll.querySelector('input').checked = false;
     } else {
       this.selectAll.querySelector('input').checked = true;
     }
+    // Display no-results message.
+    const noResultsElement = this.searchContainer.querySelector(
+      '.ecl-select__multiple-no-results'
+    );
+    if (visible.length === 0 && noResultsElement === null) {
```

Comment:
Being a parameter feeding the component, the availability of the text for the "no result" case might not be present, i think we can add a condition here so that we don't process it in case there is no label for it (unless we want to set a default on our side..)